src/HDF5.jl

```diff
@@ -984,6 +984,59 @@ t_commit(parent::Union{HDF5File, HDF5Group}, path::String, dtype::HDF5Datatype)
 
 a_create(parent::Union{HDF5File, HDF5Object}, name::String, dtype::HDF5Datatype, dspace::HDF5Dataspace) = HDF5Attribute(h5a_create(checkvalid(parent).id, name, dtype.id, dspace.id), file(parent))
 
+function _prop_get(p::HDF5Properties, name::Symbol)
+    class = p.class
+
+    if class == H5P_FILE_CREATE
+        return name === :userblock   ? h5p_get_userblock(p) :
+               name === :track_times ? h5p_get_obj_track_times(p) : # H5P_OBJECT_CREATE
+               error("unknown file create property ", name)
+    end
+
+    if class == H5P_FILE_ACCESS
+        return name === :alignment     ? h5p_get_alignment(p) :
+               name === :driver        ? h5p_get_driver(p) :
+               name === :driver_info   ? h5p_get_driver_info(p) :
+               name === :fapl_mpio     ? h5p_get_fapl_mpio(p) :
+               name === :fclose_degree ? h5p_get_fclose_degree(p) :
+               name === :libver_bounds ? h5p_get_libver_bounds(p) :
+               error("unknown file access property ", name)
+    end
+
+    if class == H5P_GROUP_CREATE
+        return name === :local_heap_size_hint ? h5p_get_local_heap_size_hint(p) :
+               name === :track_times ? h5p_get_obj_track_times(p) : # H5P_OBJECT_CREATE
+               error("unknown group create property ", name)
+    end
+
+    if class == H5P_LINK_CREATE
+        return name === :char_encoding ? h5p_get_char_encoding(p) :
+               name === :create_intermediate_group ? h5p_get_create_intermediate_group(p) :
+               error("unknown link create property ", name)
+    end
+
+    if class == H5P_DATASET_CREATE
+        return name === :alloc_time  ? h5p_get_alloc_time(p) :
+               name === :chunk       ? get_chunk(p) :
+               #name === :external    ? h5p_get_external(p) :
```

Comment:
any reason this is commented out?

Reply:
Yes — I haven't figured out how to actually correctly implement the get part of the external properties. The corresponding C calls aren't actually defined yet, and the original Dict just had `nothing` for the getter part. It looks like there's probably something useful to return, but I don't understand the external interface yet, so I just left that for a follow-up to not slow me down. (I actually have defining the corresponding bindings in a `git stash` just waiting the motivation to figure it out.)